SpatialGDK/Source/SpatialGDK/Public/Schema/ActorOwnership.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/Component.h"
+#include "SpatialCommonTypes.h"
+#include "Utils/SchemaUtils.h"
+
+#include "WorkerSDK/improbable/c_worker.h"
+
+class USpatialPackageMapClient;
+
+namespace SpatialGDK
+{
+// The ActorOwnership component defines actor's player ownership; OwnerActorEntityId
+// points to the PlayerController entity that owns a given actor.
+struct SPATIALGDK_API ActorOwnership
+{
+	static const Worker_ComponentId ComponentId = SpatialConstants::ACTOR_OWNERSHIP_COMPONENT_ID;
+
+	ActorOwnership(Worker_EntityId InLeaderEntityId)
+		: OwnerActorEntityId(InLeaderEntityId)
+	{
+	}
+
+	ActorOwnership()
+		: ActorOwnership(Worker_EntityId(SpatialConstants::INVALID_ENTITY_ID))
+	{
+	}
+
+	ActorOwnership(const ComponentData& Data) { ApplySchema(Data.GetFields()); }
+
+	ActorOwnership(const Worker_ComponentData& Data)
+	{
+		Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data.schema_type);
+
+		ApplySchema(ComponentObject);
+	}
+
+	static ActorOwnership CreateFromActor(const AActor& Actor, const USpatialPackageMapClient& PackageMap);
+
+	ComponentData CreateComponentData() const { return CreateComponentDataHelper(*this); }
+
+	ComponentUpdate CreateComponentUpdate() const { return CreateComponentUpdateHelper(*this); }
+
+	void ApplyComponentUpdate(const ComponentUpdate& Update) { ApplySchema(Update.GetFields()); }
+
+	void ApplySchema(Schema_Object* Schema)
```

Comment:
Any reason why non-trivial functions are included in this header?

Reply:
Every other handwritten schema file does the same (consider https://github.com/spatialos/UnrealGDK/blob/master/SpatialGDK/Source/SpatialGDK/Public/Schema/ServerWorker.h, https://github.com/spatialos/UnrealGDK/blob/master/SpatialGDK/Source/SpatialGDK/Public/Schema/RPCPayload.h, https://github.com/spatialos/UnrealGDK/blob/master/SpatialGDK/Source/SpatialGDK/Public/Schema/Tombstone.h).
I'd also argue that these are trivial enough to not warrant moving them into a cpp.